### DISCUSSION TITLE
SQL managed Instance -przyłączenie do AD on premise

### DISCUSSION THREAD

**Discussion by chomamirek** [AzureCommunityPL/group#370] [Pomoc]
Witam
Posiadam SQL Managed Instance i potrzebuje aby je zintegrować z Active Directory które jest w on premise a nie z Azure Active Directory. 
Czy ktoś słyszał lub wie czy tak się da czy tylko integracja z AAD jak ponizej na obrazku. 
![image](https://user-images.githubusercontent.com/35068320/138078204-5b8abef5-e33c-4df6-b199-37870c210e45.png)

Oraz sprawa numer dwa. 
Aby zabezpieczyć MI,  chcę wyłączyć reset hasła. Niestety w RBAC Custom Role nie ma opcji abym zrobił role którą wyłączam reset dla Contributorów. Czy jest jakaś opcja aby tylko Owner mógł resetować hasło do MI?

![image](https://user-images.githubusercontent.com/35068320/138078498-5d21896c-622d-4834-88b5-41154b04f270.png)
Z góry dziękuję

**Comment by huczas** [comment]
w drugim temacie to musisz zrobić całą custom role. Resetowanie hasła odbywa się przez extension w maszynie wirtualnej więc coś takiego zablokuje możliwość resetu ale i też możliwość ingerencji w inne extensiony:
`Microsoft.Compute/virtualMachines/extensions/*/write` w sekcji NotActions coś takiego:
```
{
    "Name":  "Virtual Machine Administrator No Password Reset",
    "Id":  "1234-1234-1234-1234",
    "IsCustom":  false,
    "Description":  "Virtual Machine Contributor with no Password Reset Capability",
    "Actions":  [
                    "Microsoft.Authorization/*/read",
                    "Microsoft.Compute/availabilitySets/*",
                    "Microsoft.Compute/locations/*",
                    "Microsoft.Compute/virtualMachines/*",
                    "Microsoft.Compute/virtualMachineScaleSets/*",
                    "Microsoft.DevTestLab/schedules/*",
                    "Microsoft.Insights/alertRules/*",
                    "Microsoft.Network/applicationGateways/backendAddressPools/join/action",
                    "Microsoft.Network/loadBalancers/backendAddressPools/join/action",
                    "Microsoft.Network/loadBalancers/inboundNatPools/join/action",
                    "Microsoft.Network/loadBalancers/inboundNatRules/join/action",
                    "Microsoft.Network/loadBalancers/probes/join/action",
                    "Microsoft.Network/loadBalancers/read",
                    "Microsoft.Network/locations/*",
                    "Microsoft.Network/networkInterfaces/*",
                    "Microsoft.Network/networkSecurityGroups/join/action",
                    "Microsoft.Network/networkSecurityGroups/read",
                    "Microsoft.Network/publicIPAddresses/join/action",
                    "Microsoft.Network/publicIPAddresses/read",
                    "Microsoft.Network/virtualNetworks/read",
                    "Microsoft.Network/virtualNetworks/subnets/join/action",
                    "Microsoft.RecoveryServices/locations/*",
                    "Microsoft.RecoveryServices/Vaults/backupFabrics/backupProtectionIntent/write",
                    "Microsoft.RecoveryServices/Vaults/backupFabrics/protectionContainers/protectedItems/*/read",
                    "Microsoft.RecoveryServices/Vaults/backupFabrics/protectionContainers/protectedItems/read",
                    "Microsoft.RecoveryServices/Vaults/backupFabrics/protectionContainers/protectedItems/write",
                    "Microsoft.RecoveryServices/Vaults/backupPolicies/read",
                    "Microsoft.RecoveryServices/Vaults/backupPolicies/write",
                    "Microsoft.RecoveryServices/Vaults/read",
                    "Microsoft.RecoveryServices/Vaults/usages/read",
                    "Microsoft.RecoveryServices/Vaults/write",
                    "Microsoft.ResourceHealth/availabilityStatuses/read",
                    "Microsoft.Resources/deployments/*",
                    "Microsoft.Resources/subscriptions/resourceGroups/read",
                    "Microsoft.SqlVirtualMachine/*",
                    "Microsoft.Storage/storageAccounts/listKeys/action",
                    "Microsoft.Storage/storageAccounts/read",
                    "Microsoft.Support/*"
                ],
    "NotActions":  [
		     "Microsoft.Compute/virtualMachines/extensions/*/write"
                   ],
    "DataActions":  [

                    ],
    "NotDataActions":  [```

**Reply by chomamirek** [reply]
Dzięki za podpowiedz @huczas 
Niestety to działa tylko na maszyny wirtualne. Obrazek ponizej
![image](https://user-images.githubusercontent.com/35068320/138244232-7e653110-f47a-479c-abda-524fce089b38.png)

Na SQL, Manage Instance oraz wszelkie inne nie działa i tam mozesz resetować hasła do woli 
![image](https://user-images.githubusercontent.com/35068320/138244457-a6d60152-6821-4015-a6d2-39efedd29cc7.png)

**Comment by chomamirek** [comment]
Microsoft potwierdził ze w Azure Manage Instances nie da sie przyłączyć do AD nie Azurowego oraz ze nie da się wyłącyć resetowania hasła poprzez Custom Role. 
Dla zainteresowanych to inne obejście nie koniecznie dobre. 
Tworzymy politykę która zakłada Locka ma MI a następnie wyłączamy zdejmowanie locka dla wszystkich innych niż Owner. 
I nie ma możliwości zmiany hasła. 
Minus taki ze nie można zmienić też i innych kwestii. Ale bezpieczeństwo jest najważniejsze :-)
